cmd/agent/app/reporter/grpc/reporter.go

```diff
@@ -76,20 +76,15 @@ func (r *Reporter) send(spans []*model.Span, process *model.Process) error {
 	return err
 }
 
-// addTags appends jaeger tags for the agent to every span it sends to the collector.
-func addProcessTags(spans []*model.Span, process *model.Process, agentTags []model.KeyValue) ([]*model.Span, *model.Process) {
```

Comment:
I think this method should remain here. The TagsMerger can be defined as:

```go
type TagsMerger func(spanTags, agentTags []model.KeyValue) []model.KeyValue
```

and then we simply replace the call to append() here to merger()

Reply:
Updated it, as per your suggestion